node/chainSimulator/configs/configs.go

```diff
@@ -262,3 +314,37 @@ func GetLatestGasScheduleFilename(directory string) (string, error) {
 
 	return path.Join(directory, filename), nil
 }
+
+func generateWalletKeyForShard(shardID, numOfShards uint32, converter core.PubkeyConverter) (*dtos.WalletKey, error) {
+	walletSuite := ed25519.NewEd25519()
+	walletKeyGenerator := signing.NewKeyGenerator(walletSuite)
+
+	for {
+		sk, pk := walletKeyGenerator.GeneratePair()
+
+		pubKeyBytes, err := pk.ToByteArray()
+		if err != nil {
+			return nil, err
+		}
+
+		addressShardID := shardingCore.ComputeShardID(pubKeyBytes, numOfShards)
+		if addressShardID != shardID {
+			continue
+		}
+
+		privateKeyBytes, err := sk.ToByteArray()
+		if err != nil {
+			return nil, err
+		}
+
+		address, err := converter.Encode(pubKeyBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		return &dtos.WalletKey{
+			Address:       address,
+			PrivateKeyHex: hex.EncodeToString(privateKeyBytes[:32]),
```

Comment:
is the trim really necessary?

Reply:
Yes, because the part` [32:len(privateKeyBytes] `contains the public key bytes

Reply:
but it doesn't matter, anyway